packages/react-next/src/components/Coachmark/PositioningContainer/PositioningContainer.tsx

```diff
@@ -42,397 +42,391 @@ const SLIDE_ANIMATIONS = {
   [RectangleEdge.right]: 'slideRightIn20',
 } as const;
 
-export interface IPositioningContainerState {
-  /**
-   * Current set of calcualted positions for the outermost parent container.
-   */
-  positions?: IPositionedData;
-
-  /**
-   * Tracks the current height offset and updates during
-   * the height animation when props.finalHeight is specified.
-   */
-  heightOffset?: number;
-}
-
-export class PositioningContainer extends React.Component<IPositioningContainerProps, IPositioningContainerState>
-  implements PositioningContainer {
-  public static defaultProps: IPositioningContainerProps = {
-    preventDismissOnScroll: false,
-    offsetFromTarget: 0,
-    minPagePadding: 8,
-    directionalHint: DirectionalHint.bottomAutoEdge,
-  };
-
-  private _didSetInitialFocus: boolean;
-
-  /**
-   * The primary positioned div.
-   */
-  private _positionedHost = React.createRef<HTMLDivElement>();
+const DEFAULT_PROPS = {
+  preventDismissOnScroll: false,
+  offsetFromTarget: 0,
+  minPagePadding: 8,
+  directionalHint: DirectionalHint.bottomAutoEdge,
+};
 
-  // @TODO rename to reflect the name of this class
-  private _contentHost = React.createRef<HTMLDivElement>();
+function useTargets({ target }: IPositioningContainerProps, positionedHost: React.RefObject<HTMLDivElement | null>) {
```

Comment:
This is optional, but since PositioningContainer is a fork of Callout, would it be possible to reuse any of the hooks from Callout here?

Reply:
@ecraig12345 I'll do this in a PR fixing ContextualMenu's UNSAFE methods. Since this change has a `package.json` change that causes conflicts after every publish, I'd like to merge this today.

Reply:
Sounds good.